### PR TITLE
[Telemetry Operator] Unify error handling in controllers

### PR DESCRIPTION
**Pull request by skhalash**
**Description**

Changes proposed in this pull request:

- Always return an error from the `Reconcile` function. For now, we will retry on each error, so no retry timeout has to be explicitly set
- Status update is deferred to make sure it's always executed (even if in case of an early return)
- Trigger reconciliation for referenced secret creation, not only updates

**Related issue(s)**

### CODE REVIEW

File: components/telemetry-operator/controller/logpipeline/setup.go [hunk 1]
```diff
@@ -17,32 +17,42 @@ import (
 	"sigs.k8s.io/controller-runtime/pkg/source"
 )
 
+var (
+	createOrUpdate = predicate.Funcs{
```
Comment (lindnerby) [inline components/telemetry-operator/controller/logpipeline/setup.go:21]: Let's share these as internal pkg between all three controllers

File: components/telemetry-operator/controller/logpipeline/controller.go [hunk 2]
```diff
@@ -76,49 +75,56 @@ func NewReconciler(client client.Client, config Config, prober DaemonSetProber,
...
+				err = fmt.Errorf("failed while updating status: %v: %v", statusErr, err)
+			} else {
+				err = fmt.Errorf("failed to update status: %v", statusErr)
+			}
 		}
 	}()
 
-	if err := r.ensureFinalizers(ctx, &pipeline); err != nil {
-		return ctrl.Result{Requeue: controller.ShouldRetryOn(err)}, nil
+	if err = r.ensureFinalizers(ctx, pipeline); err != nil {
```
Comment (lindnerby) [inline components/telemetry-operator/controller/logpipeline/controller.go:106]: Can you align `ensureFinalizers` with the logparser version of this? (method vs func)

**Review by lindnerby — CHANGES_REQUESTED** [review]

File: components/telemetry-operator/controller/logpipeline/controller.go (hunk 2 above)
Comment (skhalash) [reply components/telemetry-operator/controller/logpipeline/controller.go]: Sure

**Review by lindnerby — APPROVED** [review]

File: components/telemetry-operator/controller/logpipeline/setup.go (hunk 1 above)
Comment (skhalash) [reply components/telemetry-operator/controller/logpipeline/setup.go]: Sure

**Pull request merged**